src/realm/transaction.cpp

```diff
@@ -45,32 +44,71 @@ ColInfo get_col_info(const Table* table)
     return cols;
 }
 
+void add_list_to_repl(CollectionBase& list, Replication& repl, util::UniqueFunction<void(Mixed)> update_embedded);
+void add_dictionary_to_repl(Dictionary& dict, Replication& repl, util::UniqueFunction<void(Mixed)> update_embedded)
+{
+    size_t sz = dict.size();
+    for (size_t n = 0; n < sz; ++n) {
+        const auto& [key, val] = dict.get_pair(n);
+        repl.dictionary_insert(dict, n, key, val);
+        if (val.is_type(type_List)) {
+            auto n_list = dict.get_list({key.get_string()});
+            add_list_to_repl(*n_list, *dict.get_table()->get_repl(), std::move(update_embedded));
```

Comment:
You can just as well use nullptr for the function here. There will never be embedded objects in a nested collection. This comment applies to all 4 recursive calls.